app/controllers/api/v1/offers_controller.rb

```diff
@@ -49,14 +49,26 @@ def create
     param :ids, Array, of: Integer, desc: "Filter by offer ids e.g. ids = [1,2,3,4]"
     param :state, Offer.valid_states, desc: "Filter by an offer state e.g. state=draft"
     param :reviewed_by_id, String, desc: "Filter by reviewer id e.g. reviewed_by_id = 1"
+    param :category, ["finished"], desc: "To get finished(received and closed) offers"
     def index
-      @offers = @offers.with_eager_load # this maintains security
+      return finished if params["category"] == "finished"
+      @offers = params['state'] ?
```

Comment:
Avoid multi-line ?: (the ternary operator); use `if`/`unless` instead.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.